kayak_core/src/styles/mod.rs

```diff
@@ -139,21 +144,28 @@ define_styles! {
         pub background_color : StyleProp<Color>,
         pub border_color: StyleProp<Color>,
         pub border_radius: StyleProp<(f32, f32, f32, f32)>,
-        pub border: StyleProp<(f32, f32, f32, f32)>,
+        pub border: StyleProp<Edge<f32>>,
         pub bottom: StyleProp<Units>,
         pub color: StyleProp<Color>,
+        /// The spacing between child widgets along the horizontal axis
+        pub col_gap: StyleProp<Units>,
         pub cursor: StyleProp<CursorIcon>,
         pub height: StyleProp<Units>,
         pub layout_type: StyleProp<LayoutType>,
         pub left: StyleProp<Units>,
-        pub margin_bottom: StyleProp<Units>,
-        pub margin_left: StyleProp<Units>,
-        pub margin_right: StyleProp<Units>,
-        pub margin_top: StyleProp<Units>,
         pub max_height: StyleProp<Units>,
         pub max_width: StyleProp<Units>,
         pub min_height: StyleProp<Units>,
         pub min_width: StyleProp<Units>,
+        /// The inner padding between the edges of this widget and its children
+        ///
+        /// This property has lower precedence than its more specific counterparts
+        /// (`padding_top`, `padding_right`, `padding_bottom`, and `padding_left`), allowing it
+        /// to be overridden.
+        ///
+        /// A child with their own padding properties set to anything other than `Units::Auto` will
+        /// override the padding set by this widget.
+        pub padding: StyleProp<Edge<Units>>,
```

Comment:
If we want to keep `padding`, we probably want to add one that covers `top`, `right`, `bottom`, and `left` as well, right? If so, what should that be named? `offset`? `position`?

Reply:
Not sure I follow? Isn't padding top, right, bottom and left already? Or do you mean separate properties?

Reply:
> Not sure I follow? Isn't padding top, right, bottom and left already? Or do you mean separate properties?

Sorry haha yeah that wasn't very clear.

Right now `padding` covers `padding_top`, `padding_right`, `padding_bottom`, and `padding_left`. My question is, do we want something similar to cover `top`, `right`, `bottom`, and `left`?

I ask mainly because those latter properties act as both positioning and "margin" within morphorm it seems. And if we do include a property for them, should it be named something like `offset` or `position` or something else?